CI script arg parsing and compile options

anthonybilinski opened the pull request:
- [x] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/6542)
<!-- Reviewable:end -->

anthonybilinski committed:
chore(CI): Add script for handling platform compile args

anthonybilinski committed:
chore(CI): Use platform detection script in Windows dep build scripts

Allows future macOS configuration in each individual build script to not all be
duplicated. Also deduplicates current Windows arg checking and config setting.

warisb commented on buildscripts/platform_detection.sh:
```diff
@@ -0,0 +1,44 @@
+
+usage()
+{
+    echo "Download and build $DEP_NAME for the Windows cross compiling environment"
```
maybe nitpick but the filename doesn't seem to match the usage message

warisb replied:
I see it's something the dep downloader sources. Seems fine then.

warisb replied:
after offline discussion I think this file may be due for a rename - since it provides more than one function, but it is only named after one function. will approve once this is addressed.

anthonybilinski replied:
I've updated the DEP_NAME to be passed directly to parse_arch, so that it's not passing anything through global state. File also renamed from `platform_detection.sh` to `build_utils.sh`.

warisb commented on buildscripts/build_libexif.sh:
```diff
@@ -6,42 +6,21 @@
+DEP_NAME="libexif"
+parse_arch "$@"
+
+"${SCRIPT_DIR}/download/download_libexif.sh"
```
not sure if this is a level of meta-programming you are comfortable with but you can dedupe even further by using

```
download_${DEP_NAME}.sh
```
then you can move this portion to a common script. but this would mean you have to adhere to this naming convention going forward.

warisb replied:
if you do this a secondary benefit is you can have a common `download.sh` script instead of a mixture of platform detection + download script

warisb replied:
after offline discussion this is fine as is. a common download script would not add any value beyond the build_<dep>.sh script itself.

anthonybilinski commented:
Also updated to pass the supported arches from the build script to parse_arch, since not all will need to support macOS, and removed a couple of the args exported from parse_arch after rebasing the macOS stuff back up (not in PR) since any Windows args that macOS doesn't need need to be nullable from the macOS's section. This meant a little bit more arch handling in a couple of the build scripts.

warisb reviewed (approved):
Latest revision LGTM.
